### PR TITLE
Replace log4j2 with logback

### LINKED ISSUE #6837: Replace Log4j2 with Logback
### Problem

We should migrate from Log4j2 to Logback since:

* [Log4Shell](https://en.wikipedia.org/wiki/Log4Shell)
* It's not [compatible](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-with-GraalVM#logging) with native images
* It's not the default logging library in Spring, so it requires extra configuration

### Solution

* Remove `log4j2.xml` and `log4j2-test.xml`
* Remove log4j2 dependencies in Gradle
* Remove logback and slf4j exclusions in gradle
* Change `lombok.log.custom.declaration` to use slf4j
* Customize `logging.pattern.console` to match Log4j2 pattern

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

Replace log4j2 with logback

**Related issue(s)**:

Fixes #6837

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/config/LoggingFilterTest.java [hunk 1]
```diff
@@ -41,18 +35,27 @@ class LoggingFilterTest {
...
     }
 
-    @CsvSource({"/, 200, INFO", "/actuator/, 200, DEBUG"})
-    @ParameterizedTest
-    void filterOnSuccess(String path, int code, String level, CapturedOutput output) {
-        MockServerWebExchange exchange =
-                MockServerWebExchange.from(MockServerHttpRequest.get(path).build());
-        exchange.getResponse().setRawStatusCode(code);
+    @Test
+    void filterPath(CapturedOutput output) {
```
Comment (steven-sheehy) [inline hedera-mirror-graphql/src/test/java/com/hedera/mirror/graphql/config/LoggingFilterTest.java:55]: Change `/actuator/` test to set level to debug and verify it gets printed to keep it at info and verify it does not get printed.

**Comment by steven-sheehy** [comment]
Note: Code smells are pre-existing

**Review by jascks — APPROVED** [review]
Looks good! I do like the colors. Verify readable and nothing frivolous about it. 👍

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**